pulp_ansible/app/tasks/collections.py

```diff
@@ -529,6 +544,7 @@ async def _read_from_downloaded_metadata(self, name, namespace, requirement):
                         )
                     )
                 )
+        await asyncio.gather(*tasks)
```

Comment:
The dependencies wouldn't be synced for the v3 codepath without this gather here. Not sure what other potential bugs were present before hand.

Reply:
If this is an unrelated bugfix can it receive it's own issue and separate commit please. That's important in cases where we say need to cherry pick back the bugfix by itself to an earlier branch.